Feature/lin230 umbrella selector

DelusionCB opened the pull request:
# LIN230 Umbrella Selectors changes

## Updated UmbrellaSelector, changed functionality, added third option called 'is_independent' & changed checkboxes to radios

### [LIN230](https://trello.com/c/Xt31muUf/230-ux-kattotapahtuma-valinta)

-----------------------------------------------------------------------------------------------
### Breakdown:

#### Umbrella
 1. src/components/HelFormFields/UmbrellaSelector/
     * Deleted UmbrellaCheckbox.js & changed UmbrellaCheckbox.scss name to UmbrellaRadio.scss
     * Added UmbrellaRadio.js to replace checkbox.
   
 2. UmbrellaSelector/UmbrellaRadio.js 
     * Changes compared to checkbox: No UncontrolledTooltip, instead of props.children, we use FormattedMessage & pass ID to it with messageID-prop.
    
 3. UmbrellaSelector/UmbrellaSelector.js 
     * Updated logic of **"handleCheck"** -function. Now more streamlined with dispatches. Uses array & object which are controlled by value, then passed to setState & clearValue.
     * Swapped UmbrellaCheckboxes to UmbrellaRadios & added 'is_independent' -radio, to work as "blank" option.
     * Updated commenting from checkboxes to radio
     * Updated **"getDisabledState"** -functions logic. Added support for 'is_independent' -value.
     
     
#### Tests & styles
 1. UmbrellaRadio.test.js & UmbrellaSelector.test.js & editorActions.test.js
     * Added testing
   
 2. Styling
     * Globally changed var(--danger-color) to var(--red-color) -> Contrast wasn't enough on danger-color
    
 3. Localization
      * Updated loca & deleted old ones
      
# Major changes after PR was created via commits
  1. formDataMapping.js
     * Changed updateSubEventsFromFormValues -functions keysToUpdate to keysToExclude, as these are values that are NOT updated to children from parent recurring & added 'sub_event_type' into exclude as we didn't want  series events childrens to inherit 'sub_umbrella' from their parent.
  2. UmbrellaSelector/UmbrellaSelector.js 
     * Added conditionals in **"handleCheck"** -function based on values.super_event_type. Added comments for function
     * Removed state-based control from **"getDisabledState"**-function. We have values to use, as state-control wouldn't work with radios as it worked with checkboxes. Also updated comments for the function, including ruleset for 'is_independent'

DelusionCB committed:
WIP commit

1. Removed UmbrellaCheckbox
2. Created UmbrellaRadio to replace checkbox
3. Updated functions in UmbrellaSelector
WIP

DelusionCB committed:
Updates to UmbrellaSelector & tests

1. Changed logic of handleCheck in UmbrellaSelector.
2. Deleted tooltips from UmbrellaRadio
3. Changed loca
4. Updated tests

DelusionCB committed:
Update UmbrellaSelector.test.js

Fixed linting

DelusionCB committed:
Added test for UmbrellaRadio.js & changed small things

1. Added test for UmbrellaRadio
2. Updated clearValue-functions comment
3. Updated tests
4. Deleted non-used locas

DelusionCB committed:
Updated tests & styles

DelusionCB committed:
Removed state which doesn't exist

DelusionCB committed:
Cleared props

DelusionCB committed:
Removed more useless props

DelusionCB committed:
Removed values to clear from has_umbrella

DelusionCB committed:
Updated logic of handleCheck -function & updated tests

Added conditionals in handlecheck based on values.super_event_type. Added comments for function

DelusionCB committed:
Update formDataMapping.js

Updated updateSubEventsFromFormValues
Added 'sub_event_type' into keysToUpdate which are not updated while super recurring is updated

aceViilee reviewed (commented):
UmbrellaSelector is a little bit messy. Should consider at some point to rewrite the logic. Perhaps chop this to smaller reusable segment.

DelusionCB committed:
Update formDataMapping.js

Changed updateSubEventsFromFormValues -functions keysToUpdate to keysToExclude, as these are values that are NOT updated to children from parent recurring

DelusionCB committed:
Updated logic & commenting of getDisabledState

Removed state-based control from getDisabled-function. We have values to use, as state-control wouldn't work with radios as it worked with checkboxes. Also updated comments for the function, including ruleset for 'is_independent'
